### ISSUE TITLE
Bluesky: GIF lost on a reply on Mastodon

### ISSUE THREAD

**Issue by dploeger** [snarfed/bridgy-fed#1113]
In [this reply](https://mastodon.social/@dploeger/112563888619602124) to a Bluesky user, the GIF wasn't available on Bluesky. Only the user handle was visible.

Sending a normal picture in a reply works though.

**Comment by snarfed** [comment]
Yes! Bluesky doesn't support GIFs as normal images, it has some kind of custom thing. Maybe Tenor specific? I haven't dug into it yet. First step here would be to find docs and understand what that custom GIF support is and how to integrate with it.

(BTW @dploeger this is the repo for Bridgy Fed bugs, not https://github.com/snarfed/bridgy . That repo is for Bridgy classic, https://brid.gy/ , a separate service. https://brid.gy/about#bridgy-fed )

**Comment by dploeger** [comment]
Oh sorry. I was confused with the two. Thanks for investigating!

**Comment by qazmlp** [comment]
The gist is that Bluesky currently doesn't support custom gifs. (Tenor is essentially a searchable meme-library that users can pick existing stuff from.)

It's possible to submit gifs to Tenor off-site, but I'm pretty sure that doesn't mesh well with retaining control or even attribution over content.

**Comment by dploeger** [comment]
I actually thought the GIF would just be an attachment. Apparently it's way more complex than that.

**Comment by qazmlp** [comment]
I assume it will be an attachment, eventually, whenever Bluesky adds actual video support. For now it only has meme-support though.

(I checked the [Tenor ToS](https://tenor.com/legal-terms) and I don't think this is compatible with Bridgy in terms of requirements and what it does with the uploads.)

**Comment by dploeger** [comment]
Ohhh okay.

**Comment by qazmlp** [comment]
I'm also thinking "no" since that'd enable IP grabbers.
(The frontend loads these embeds directly from the target server.)

Maybe you could get them to allow-list Bridgy Fed for embeds though?
The main downside is you'd potentially get hit with **a lot** of traffic due to the lack of cache there.

**Comment by snarfed** [comment]
Heard from the Bluesky team on Discord, sounds like the answer is no.

> basically our GIFs have to use a very specific tenor media URL, and also include the height and width of gif in the url search params

As for allowlisting, BF doesn't currently host media, so they couldn't easily whitelist it. I could serve redirects, but that's only minimally useful, if at all.

**Comment by snarfed** [comment]
Right. I wouldn't plan to proxy the connections for those media files anyway, I'd just redirect. Regardless, all purely hypothetical right now.

**Comment by snarfed** [comment]
Related: #1317

**Comment by danielhatton** [comment]
What about the converse?  If a BlueSky user posts an animation in whatever format BlueSky does support, would you expect that to be successfully bridged to fedi?

**Comment by snarfed** [comment]
@danielhatton not right now. The issue is basically the same in both directions; Bluesky's GIFs are proprietary, via Tenor, and don't Just Work like normal images, videos, or external embeds (ie link previews). Details above.

**Comment by snarfed** [comment]
Evidently new "custom" GIFS are converted to videos and uploaded as blobs: https://discord.com/channels/1097580399187738645/1097580399187738648/1414724731764084786 ...which is nice here, since we'll support them like normal videos.
